packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -662,4 +663,54 @@ export class Cluster extends ClusterBase {
       throw new Error('Cannot add a parameter to an imported parameter group.');
     }
   }
+
+  /**
+   * Adds a role to the cluster
+   *
+   * @param role the role to add
+   */
+  public addIamRole(role: iam.IRole): void {
+    const clusterRoleList = this.cluster.iamRoles ?? [];
+
+    if (clusterRoleList.includes(role.roleArn)) {
+      throw new Error('Role is already attached to the cluster');
+    }
+    if (clusterRoleList.length >= 10) {
+      throw new Error('Maximum number of IAM roles for a cluster is 10');
```

Comment:
According to https://docs.aws.amazon.com/redshift/latest/mgmt/amazon-redshift-limits.html, the limit is 50 in most regions.

I'd rather not do region-specific weirdness, and instead just let it default to 50.